### PR TITLE
#2954 Can't connect to custom local interpreter

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
- Pipe must be opened in async mode for the WaitForConnectionAsync to be cancellable. Otherwise token is ignored.
- We need to cancel on either process exit or user clicks cancel
- Actual issue is that broker derives log name from the connection name. Hence we need to make sure the name is reasonably clean.
- Also fixed the issue when typing bad connection names crashed VS

### CODE REVIEW

File: src/Mocks/VisualStudio/project.15.0.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "Microsoft.VisualStudio.Editor": "15.0.25824-RC",
     "Microsoft.VisualStudio.ImageCatalog": "15.0.25824-RC",
     "Microsoft.VisualStudio.ProjectSystem": "15.0.594-pre",
-    "Microsoft.VisualStudio.Imaging.Interop.14.0.DesignTime": "14.3.25407",
```
Comment (int19h) [inline src/Mocks/VisualStudio/project.15.0.json]: Probably wrong.

File: src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs [hunk 2]
```diff
@@ -192,7 +194,7 @@ internal sealed class ConnectionViewModel : BindableBase, IConnectionViewModel {
 
             Uri uri;
             var isPathValid = Uri.TryCreate(Path, UriKind.Absolute, out uri);
-            if (string.IsNullOrEmpty(Name)) {
+            if (!IsValidConnectionName(Name)) {
```
Comment (int19h) [inline src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs]: We should also validate this in broker side, and reject invalid names in some sensible way (i.e. as API error).

File: src/Mocks/VisualStudio/project.15.0.json (hunk 1 above)
Comment (MikhailArkhipov) [inline src/Mocks/VisualStudio/project.15.0.json]: Weird. I have `    "Microsoft.VisualStudio.Imaging.Interop.14.0.DesignTime": "15.0.25726-Preview5"` locally...

Comment (MikhailArkhipov) [inline src/Mocks/VisualStudio/project.15.0.json]: That's 15.0 and what do you know, `15.0.25726-Preview5` is the latest. go figure.

File: src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs (hunk 2 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs]: It fails to start the process, throws in n`ew FileLoggerProvider(_startupOptions.Name)` so there is no connection to report the API error.

**Comment by int19h** [comment]
👍

**Pull request merged**